src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerFindMatches.java

```diff
@@ -0,0 +1,161 @@
+package uk.gov.hmcts.divorce.caseworker.event;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.elasticsearch.index.query.BoolQueryBuilder;
+import org.elasticsearch.index.query.QueryBuilders;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.sdk.api.CCDConfig;
+import uk.gov.hmcts.ccd.sdk.api.CaseDetails;
+import uk.gov.hmcts.ccd.sdk.api.ConfigBuilder;
+import uk.gov.hmcts.ccd.sdk.api.callback.AboutToStartOrSubmitResponse;
+import uk.gov.hmcts.ccd.sdk.type.CaseLink;
+import uk.gov.hmcts.ccd.sdk.type.ListValue;
+import uk.gov.hmcts.divorce.common.ccd.PageBuilder;
+import uk.gov.hmcts.divorce.divorcecase.model.Application;
+import uk.gov.hmcts.divorce.divorcecase.model.CaseData;
+import uk.gov.hmcts.divorce.divorcecase.model.CaseMatch;
+import uk.gov.hmcts.divorce.divorcecase.model.MarriageDetails;
+import uk.gov.hmcts.divorce.divorcecase.model.State;
+import uk.gov.hmcts.divorce.divorcecase.model.UserRole;
+import uk.gov.hmcts.divorce.idam.IdamService;
+import uk.gov.hmcts.divorce.systemupdate.service.CcdSearchService;
+import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Map;
+
+import static uk.gov.hmcts.divorce.divorcecase.model.State.POST_SUBMISSION_STATES;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.CASE_WORKER;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.CASE_WORKER_BULK_SCAN;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.SUPER_USER;
+import static uk.gov.hmcts.divorce.divorcecase.model.access.Permissions.CREATE_READ_UPDATE;
+import static uk.gov.hmcts.divorce.notification.FormatUtil.ES_DATE_FORMATTER;
+import static uk.gov.hmcts.divorce.systemupdate.service.CcdSearchService.REFERENCE_KEY;
+import static uk.gov.hmcts.divorce.systemupdate.service.CcdSearchService.STATE_KEY;
+
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class CaseworkerFindMatches implements CCDConfig<CaseData, State, UserRole> {
+
+    public static final String FIND_MATCHES = "caseworker-find-matches";
+    private final CcdSearchService ccdSearchService;
+    private final IdamService idamService;
+    private final AuthTokenGenerator authTokenGenerator;
+
+    protected final ObjectMapper objectMapper;
+
+    @Override
+    public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuilder) {
+
+        new PageBuilder(configBuilder
+            .event(FIND_MATCHES)
+            .forStates(POST_SUBMISSION_STATES)
```

Comment:
Do you think they might want to use this for pre-submission cases? We often seem to have cases stuck in draft when an app has created multiple cases

Reply:
It's a check to find live cases in progress to make sure same divorce doesn't go in front of multiple judges I think.